src/status_im/common/device_network.cljs

```diff
@@ -0,0 +1,55 @@
+(ns status-im.common.device-network
+  (:require
+    ["@react-native-community/netinfo" :default net-info]
+    [native-module.core :as native-module]
+    [status-im.feature-flags :as ff]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(rf/reg-fx
+ :effects.device-network/listen-to-network-info
+ (fn []
+   (when net-info
+     (.addEventListener ^js net-info
+                        #(rf/dispatch [:device-network/on-state-change
+                                       (js->clj % :keywordize-keys true)])))))
+
+(rf/reg-event-fx
+ :device-network/on-state-change
+ (fn [{:keys [db]} [{:keys [isConnected type details]}]]
+   (let [old-network-status       (:network/status db)
+         old-network-type         (:network/type db)
+         connectivity-status      (if isConnected :online :offline)
+         status-changed?          (not= connectivity-status old-network-status)
+         type-changed?            (not= type old-network-type)
+         is-connection-expensive? (:is-connection-expensive details)]
+     (log/debug "[net-info]"
```

Comment:
I don't know if it's important, but sharing this handler is very similar to the one in https://github.com/status-im/status-mobile/pull/20730/files#diff-efce04a1c18c0acfd8ece4416c70f868204e0a4557c5f649bea69d1c4311380cR44

Reply:
ahh what are the odds 😆

Reply:
should've merged when I had the chance :). This is the same change as there, basically just moved the legacy code to status-im. I'll check with @Parveshdhull.